### ISSUE TITLE
Some issues about cuboid compactness loss

### ISSUE THREAD

**Issue by duzhenjiang113** [SilenKZYoung/CuboidAbstractionViaSeg#3]
Hi, Kai Zhi, great job. And I have a question about compactness loss after reading the paper. M is a hyperparameter and fixed, why should we use a loss function to reduce the number of parts? 
Thanks again, looking forward to your reply.

**Comment by SilenKZYoung** [comment]
Hi, about this question, we can consider M as the number of types of parts that appear in all shapes of a category. However, not all types of parts appear for a particular shape, so we do not use all M cubes when representing a particular shape. While the human perception of structure is concise, we expect both to use as few cubes as possible (compactness loss) to represent the object and approximate the original geometry and possible (reconstruction loss).
